common/app/views/fragments/image.scala.html

```diff
@@ -10,6 +10,19 @@
     isImmersiveMainMedia: Boolean = false
 )
 
+@immersivePortraitSource(largestImage: model.ImageAsset, hidpi: Boolean = false) = {
+    <source media="@if(hidpi) {
+        (orientation: portrait) and (-webkit-min-device-pixel-ratio: 1.25), (orientation: portrait) and (min-resolution: 120dpi)
+    } else {
+        (orientation: portrait)
+    }"
+    sizes="@{100 * (largestImage.width.toDouble / largestImage.height.toDouble)}vh"
+    srcset="@{(500 to 4000 by 250)
```

Comment:
The number of provided images here is totally up for question. In an ideal world we wouldn't have to list specific breakpoints like this, the browser would download the exact size using [client hints](https://developers.google.com/web/updates/2015/09/automating-resource-selection-with-client-hints), but we're not there yet. (Maybe we'll never be due to cache fragmenting issues?) In the interim, the lower the interval we set here (currently 250) the better for the user as it will provide a closer match without rounding up/down to a much smaller/bigger image.

@paperboyo I think the browser rounds to the nearest image, e.g. a screen size of 600px would download the 500px image, whereas a screen size of 700px would download the 750px image. We should test it!